ModelBase/src/nodes/Reference.cpp

```diff
@@ -85,7 +85,8 @@ bool Reference::resolveHelper(bool indirect)
 
 	auto newTarget = computeTarget();
 
-	Q_ASSERT(!newTarget || (newTarget->definesSymbol() && newTarget->symbolName() == name_));
+	//TODO@cyril What is this for?
+//	Q_ASSERT(!newTarget || (newTarget->definesSymbol() && newTarget->symbolName() == name_));
```

Comment:
Oh man, this is embarrassing. You are absolutely correct.

Anyway, now that I think of it, we don't even need to use the ATTRIBUTES. Just create a regular, plain-old Node* filed and have a way of setting it. The fact that we don't really need to persist these nodes using the standard mechanism, let's us do pretty much whatever we want.

We just need to be careful when creating the visualization, that we don't assume this Node* field is part of our own tree, but I think this will happen naturally.